### PR TITLE
Zqs 832/updated problem docstrings

### PR DESCRIPTION
**Pull request by AthenaCaesura**
3 changes in this branch:
1. Updated docstrings in problems folder to reflect current code and added descriptions of problem being solved.
2. Changed StableSet to MaxIndependentSet wherever necessary.
3. Reconfigured `_hamiltonian_factor` in vertex_cover.py  to `_A` and `_B` to better reflect equations in source material. Also updated docstring in vertex_cover.py to better explain the roles of `_A` and `_B`.

### CODE REVIEW

File: src/python/zquantum/qaoa/problems/graph_partition.py [hunk 1]
```diff
@@ -5,6 +5,17 @@
 
 
 class GraphPartitioning(Problem):
+    """Solves the graph partitioning problem for undirected, unweighted graph using
+    an ising model formulation.
+
+    The solution of a graph partitioning  problem is the set of nodes S that minimizes
+    the wieght of the edges between S and the rest of the graph. S must contain half
```
Comment (mstechly) [inline src/python/zquantum/qaoa/problems/graph_partition.py:12]:
```suggestion
    the weight of the edges between S and the rest of the graph. S must contain half
```

File: src/python/zquantum/qaoa/problems/graph_partition.py [hunk 2]
```diff
@@ -5,6 +5,17 @@
...
+    the nodes in the graph.
+
+    From "Ising formulations of many NP problems" by A. Lucas, page 6
```
Comment (mstechly) [inline src/python/zquantum/qaoa/problems/graph_partition.py:15]:
```suggestion
    From "Ising formulations of many NP problems" by A. Lucas, page 6, eqs 7-9
```

File: src/python/zquantum/qaoa/problems/vertex_cover.py [hunk 3]
```diff
@@ -5,42 +5,48 @@
...
+    formulation.
+
+    The solution of a vertex cover problem is the minimal number of colored
+    verticies such that all edges connect to a colored vertex.
+    From "Ising formulations of many NP Problems" by A. Lucas, eq. 33 and 34
+    (https://arxiv.org/pdf/1302.5843.pdf)
+    and
+    https://quantumcomputing.stackexchange.com/questions/16082/vertex-cover-mappings-from-qubo-to-ising-and-vice-versa
+    for corrective translation shifts
+
```
Comment (mstechly) [inline src/python/zquantum/qaoa/problems/vertex_cover.py:18]: Add info about args here as well.
Put explanation of `_A` and `_B` to args, and refer to that in docs for `_A` and `_B`.

**Review by mstechly — APPROVED** [review]

File: src/python/zquantum/qaoa/problems/vertex_cover.py (hunk 3 above)
Comment (AthenaCeasura) [reply src/python/zquantum/qaoa/problems/vertex_cover.py:18]: Made the suggested changes.

**Pull request merged**